Export inrangecount from NearestNeighbors.jl

kahaaga opened the pull request:
Export `inrangecount` from NearestNeighbors.jl, as discussed in https://github.com/JuliaDynamics/Entropies.jl/pull/71#discussion_r1002435920 and in #16.

Is this approach ok? I'm not sure whether to export `inrangecount` directly, or override it with its own docstring here.

kahaaga committed:
Export inrangecount from NearestNeighbors.jl

kahaaga committed:
Custom function with its own docstring

Datseris commented:
i's okay but also bump patch version!

Datseris commented:
Hm, actually it isn't. This should be defined in the API file with a "search strcture" terminology. Then, the file that extends the API based on NearestNeighbors.jl should provide the concrete implementation.

kahaaga commented:
> Hm, actually it isn't. This should be defined in the API file with a "search strcture" terminology

Defining e.g. `struct WithinRangeCount; r; end` in `api.jl`, you mean? And then in `kdtree.jl` define the following?

```julia
function Neighborhood.inrangecount(tree::KDTree, querypt, t::WithinRangeCount)
    ....
end
```

kahaaga committed:
Use api

kahaaga committed:
Add tests

kahaaga committed:
Correct import location and generic syntax

kahaaga committed:
Reorganize tests

kahaaga committed:
Document WithinRangeCount

kahaaga committed:
Missing bracket

kahaaga commented:
@Datseris Ok, I think I got what you meant, and updated the PR accordingly. Changes:

- In `api.jl`: Added the `WithinRangeCount` search type, the `inrangecount` function.
- In `kdtree.jl`: Dispatch on `inrangecount(::KDTree, query, ::WithinRangeCount)`.
- Added entries for `WithinRangeCount` and `inrangecount` in the documentation.
 - Added some tests.

Datseris committed:
remove WithinRangeCount struct

Datseris commented:
That's pretty much what I've meant. However I removed the type because, by the definition of its name, `inrangecount` always uses the `WithinRange` type of search, so an additional struct is not necessary.

Datseris committed:
fix tests

kahaaga commented:
Nice! I'll update the open PRs in Entropies.jl to use the new version of Neighborhood.jl (I assume this is merged and tagged within not-too-long).
